src/daft-logical-plan/src/optimization/optimizer.rs

```diff
@@ -149,6 +146,20 @@ impl Default for OptimizerBuilder {
                     vec![Box::new(PushDownLimit::new())],
                     RuleExecutionStrategy::FixedPoint(Some(3)),
                 ),
+                // --- Rewrite projections ---
+                // Once optimization rules have been applied,split actor pool projects and detect monotonic IDs.
+                // By delaying these rewrite rules, we avoid having to special case optimization rules for
+                // actor pool projects and monotonically increasing ids.
+                RuleBatch::new(
+                    vec![
+                        Box::new(SplitActorPoolProjects::new()),
+                        // Push down projections after splitting actor pool projects.
+                        Box::new(PushDownProjection::new()),
```

Comment:
Could we move this under all of the projection rewrite rules? Maybe we could even make this its own rule batch, the other rules do not need to be run more than once.